awx/main/tests/functional/api/test_job_runtime_params.py

```diff
@@ -370,6 +370,52 @@ def test_job_launch_fails_with_missing_vault_password(machine_credential, vault_
     assert response.data['passwords_needed_to_start'] == ['vault_password']
 
 
+@pytest.mark.django_db
+def test_job_launch_fails_with_missing_multivault_password(machine_credential, vault_credential,
+                                                           deploy_jobtemplate, get, post, rando):
+    vault_cred_first = Credential(
+        name='Vault #1',
+        credential_type=vault_credential.credential_type,
+        inputs={
+            'vault_password': 'ASK',
+            'vault_id': 'abc'
+        }
+    )
+    vault_cred_first.save()
+    vault_cred_second = Credential(
+        name='Vault #2',
+        credential_type=vault_credential.credential_type,
+        inputs={
+            'vault_password': 'ASK',
+            'vault_id': 'xyz'
+        }
+    )
+    vault_cred_second.save()
+    deploy_jobtemplate.credentials.add(vault_cred_first)
+    deploy_jobtemplate.credentials.add(vault_cred_second)
+    deploy_jobtemplate.execute_role.members.add(rando)
+    deploy_jobtemplate.save()
+
+    url = reverse('api:job_template_launch', kwargs={'pk': deploy_jobtemplate.pk})
+    resp = get(url, rando, expect=200)
+    assert resp.data['passwords_needed_to_start'] == ['vault_password.abc', 'vault_password.xyz']
+
+    resp = post(url, rando, expect=400)
+    assert resp.data['passwords_needed_to_start'] == ['vault_password.abc', 'vault_password.xyz']
```

Comment:
Note for future direction: Do we really want to test `passwords_needed_to_start` anymore? The discussions I had with the UI pointed toward using the passwords needed inside of the credentials in `defaults`. So we might consider going back over all the tests and tearing it out of the API functional tests.

Reply:
I think for now it's probably reasonable to test both since we support both.